test/e2e/catalogutil/catalog.go

```diff
@@ -0,0 +1,1537 @@
+	log(string(cmdOutputBytes))
+}
+
+// setOPMDebug is a helper function to add the debug flag to an opm executable command
+func setOPMDebug(cmd *exec.Cmd) {
+	cmd.Args = append(cmd.Args, "--debug")
+}
+
+/*
+WriteAnnotationsYaml creates an annotations.yaml file with a combination of default values and the provided arguments
+
+Arguments:
+
+• destinationDir: The metadata directory where the file will be written to
+
+• packageName: the OLM package name used in the package label
+
+• channels: a slice of channels used in the channels label
+
+• channelDefault: the default channel to be used in the default channel label
+
+Returns:
+
+error is non nil if anything prevented the bundle image from being added to the database
+*/
+func WriteAnnotationsYaml(destinationDir string, packageName string, channels []string, channelDefault string) error {
+	yamlBytes, err := bundle.GenerateAnnotations(bundle.RegistryV1Type, bundle.ManifestsDir, bundle.MetadataDir, packageName, strings.Join(channels, ","), channelDefault)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(destinationDir, bundle.AnnotationsFile), yamlBytes, os.FileMode(0644))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// createCRD is a helper function that creates a CRD using either v1 or v1beta1 apis
+func createCRD(ownedCRD CRDInformation, versions interface{}) (interface{}, error) {
+
+	switch versions.(type) {
+	case []apiextensionsv1.CustomResourceDefinitionVersion:
+		return createV1CRD(ownedCRD, versions.([]apiextensionsv1.CustomResourceDefinitionVersion)), nil
+	case []apiextensionsv1beta1.CustomResourceDefinitionVersion:
+		return createV1Beta1CRD(ownedCRD, versions.([]apiextensionsv1beta1.CustomResourceDefinitionVersion)), nil
+	default:
+		return nil, errors.New("Incorrect datatype for versions argument. Must be []apiextensionsv1.CustomResourceDefinitionVersion or []apiextensionsv1beta1.CustomResourceDefinitionVersion")
+	}
+}
+
+// createV1CRD creates a v1 CRD
+func createV1CRD(ownedCRD CRDInformation, versions []apiextensionsv1.CustomResourceDefinitionVersion) *apiextensionsv1.CustomResourceDefinition {
+	return &apiextensionsv1.CustomResourceDefinition{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "CustomResourceDefinition",
+			APIVersion: apiextensionsv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ownedCRD.Description.Name,
+		},
+		Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+			Group: ownedCRD.Group,
+			Names: apiextensionsv1.CustomResourceDefinitionNames{
+				Kind:     ownedCRD.Description.Kind,
+				Plural:   ownedCRD.PluralName,
+				Singular: ownedCRD.SingluarName,
+			},
+			Scope:    apiextensionsv1.NamespaceScoped,
+			Versions: versions,
+		},
+	}
+}
+
+// createV1Beta1CRD creates a v1beta1 CRD
+func createV1Beta1CRD(ownedCRD CRDInformation, versions []apiextensionsv1beta1.CustomResourceDefinitionVersion) *apiextensionsv1beta1.CustomResourceDefinition {
```

Comment:
We probably don't want to create any new v1beta1 CRDs at all since they are being deprecated in kube 1.22. 

We need to overhaul our own e2e tests to move to v1 CRDs entirely at some point in the near future

Reply:
I think the idea for using v1beta1 is to handle the complexity of what happens when we migrate from one to the other when your operator gets upgraded. There's a whole series of test cases we were thinking of creating around this.